pyanaconda/ui/gui/spokes/custom_storage.py

```diff
@@ -1569,22 +1580,35 @@ def on_raid_level_changed(self, widget):
         self.on_value_changed()
 
     def on_size_changed(self, widget):
+        """Callback for text change in "desired capacity" widget"""
         if not self._sizeEntry.get_sensitive():
             return
 
-        current_size = Size(self._request.device_size)
-        displayed_size = current_size.human_readable(max_places=self.MAX_SIZE_PLACES)
-
-        if displayed_size == self._sizeEntry.get_text():
-            return
-
         size = get_size_from_entry(
             self._sizeEntry,
             lower_bound=self.MIN_SIZE_ENTRY,
             units=SIZE_UNITS_DEFAULT
         )
 
+        # Show warning if the size string is invalid. Field self._error is used as a "flag" that
+        # the last error was the same. This is done because this warning can fire on every change,
+        # so it would keep flickering at the bottom as you type.
         if size is None:
+            if self._error != self.DESIRED_CAPACITY_ERROR:
+                self.clear_info()
+                self.set_detailed_warning(
```

Comment:
This solution is somehow less readable. I guess we need to somehow combine `show_warning_message` with `set_detailed_warning`.